chat-plugins/calculator.js

```diff
@@ -0,0 +1,126 @@
+'use strict';
+
+/** replace all fx */
+function replaceAll(haystack, needle, replace) {
+	return haystack.split(needle).join(replace);
+}
+
+/** standardize string format */
+function reformat(s) {
+	s = s.toLowerCase();
+	s = replaceAll(s, "-(", "-1*(");
+	s = replaceAll(s, ")(", ")*(");
+	s = replaceAll(s, " ", "");
+	s = replaceAll(s, "-", "+-");
+	s = replaceAll(s, "--", "+");
+	s = replaceAll(s, "++", "+");
+	s = replaceAll(s, "(+", "(");
+	let i;
+	for (i = 0; i < 10; i++) {
+		s = replaceAll(s, i + "(", i + "*" + "(");
+	}
+	while (s.charAt(0) === "+") s = s.substr(1);
+	return s;
+}
+
+/** determine if char should be added to side */
+function isParseable(n, minus) {
+	return (!isNaN(n) || (n === "-" && !minus) || n === ".");
+}
+
+/** general fx to get two terms of any fx (multiply, add, etc) */
+function getSide(haystack, middle, direction, minus) {
+	let i = middle + direction;
+	let term = "";
+	let limit = (direction === -1) ? 0 : haystack.length; // set the stopping point, when you have gone too far
+	while (i * direction <= limit) { // while the current position is >= 0, or <= upper limit
+		if (isParseable(haystack[i], minus)) {
+			if (direction === 1) term = term + haystack[i];
+			else term = haystack[i] + term;
+			i += direction;
+		} else { return term; }
+	}
+	return term;
+}
+
+/** fx to generically map a symbol to a function for parsing */
+function allocFx(eq, symbol, alloc, minus) {
```

Comment:
Your confusingly-named `minus` parameter (I think its intended meaning is `excludeMinusSign`) appears to do nothing, because your `reformat` function already removes all unary minus operators.

Reply:
Fixed.